subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ResolveTaskOutputCachingStateExecuter.java

```diff
@@ -51,6 +52,8 @@ public void execute(TaskInternal task, TaskStateInternal state, TaskExecutionCon
         } else {
             state.setTaskOutputCaching(DefaultTaskOutputs.DISABLED);
         }
-        delegate.execute(task, state, context);
+        if (delegate != null) {
+            delegate.execute(task, state, context);
+        }
```

Comment:
this is now moot given we went back to the old decorator pattern, with a subsection of the task executers being wrapped in a build operation